cmd/config-common.go

```diff
@@ -107,10 +108,21 @@ func readConfigEtcd(ctx context.Context, client *etcd.Client, configFile string)
 func watchConfigEtcd(objAPI ObjectLayer, configFile string, loadCfgFn func(ObjectLayer) error) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultContextTimeout)
 	defer cancel()
-	for watchResp := range globalEtcdClient.Watch(ctx, configFile) {
-		for _, event := range watchResp.Events {
-			if event.IsModify() || event.IsCreate() {
-				loadCfgFn(objAPI)
+	for {
+		select {
+		case <-GlobalServiceDoneCh:
+			return
+		case watchResp := <-globalEtcdClient.Watch(ctx, iamConfigPrefix):
+			if err := watchResp.Err(); err != nil {
+				logger.LogIf(ctx, err)
+				// log and retry.
+				time.Sleep(1 * time.Second)
```

Comment:
@harshavardhana , can you move time.Sleep(1 * time.Second) before the select - even when etcd is up,we need to make sure we don't keep polling watch

Reply:
We will never re-trigger until the watch event is invoked, watch is just a channel which will wait until it gets invoked.